src/components/PostCard/PostCard.tsx

```diff
@@ -0,0 +1,50 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Post } from '../../types/Post';
+
+type Props = {
+  post: Post;
+  loadPost: (id: number) => void;
+  setSelectedPost: (post: Post | null) => void;
+  selectedPost: Post | null,
+};
+
+export const PostCard: React.FC<Props> = ({
+  post,
+  loadPost,
+  setSelectedPost,
+  selectedPost,
+}) => {
+  const btnTitle = selectedPost?.id !== post.id ? 'Open' : 'Close';
+
+  const openPostHandler = (postId: number) => {
+    if (btnTitle === 'Open') {
+      loadPost(postId);
+    } else {
+      setSelectedPost(null);
+    }
+  };
```

Comment:
When a post is open and I click to open another one, old comments, the form and write-a-comment button should disappear